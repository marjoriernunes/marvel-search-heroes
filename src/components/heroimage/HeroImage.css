.gradient-overlay {
    position: relative;
}

.gradient-overlay ::after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-image: linear-gradient(133deg, rgba(131,58,180,1) 0%, rgba(253,29,29,1) 50%, rgba(252,176,69,1) 100%);
    opacity: 0.7;
}

.gradient-overlay img {
    filter: grayscale(1);
}

.header-hero-info {    
    position: relative;
    text-align: center;
}

.header-hero-info img{
    width: 100%;
    background: cover;
    height: 500px;
    object-fit: cover;
}

.top-left {
    position: absolute;
    top: 8px;
    width: 50%;
    text-align: left;
    color: #e4e4e4;
}

.hero-name {
    font-size: xxx-large;
    font-weight: 500;
    padding-bottom: 1rem;
    margin: 1rem;
    border-bottom: 3px solid #e4e4e4;
}

.hero-description {
    margin: 1rem;
    font-size: x-large;
    font-weight: 300;
}

@media (max-width: 860px) {
    .top-left {
        width: 100%;
    }
}

@media (max-width: 460px) {
    .top-left {
        position: relative;
        top: 8px;
        width: 100%;
        text-align: left;
        color: #404040;
    }

    .hero-name {
        border-bottom: 3px solid #404040;
    }
}