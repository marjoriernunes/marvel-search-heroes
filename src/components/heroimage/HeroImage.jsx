import './HeroImage.css';

const HeroImage = (props) => {
    return (
        <> 
            <div className="header-hero-info">
                <div className="gradient-overlay">
                    <img 
                        src={`${props.heroInfo?.thumbnail.path}/detail.${props.heroInfo?.thumbnail.extension}`}  
                        alt={props.heroInfo?.name} />
                        <span/>
                </div>
                <div className="top-left">
                   <p className="hero-name">
                        {props.heroInfo?.name}
                    </p>
                    <p className="hero-description">
                        {props.heroInfo?.description === '' 
                        ? 'Agente Nicholas Joseph "Nick" Fury está em uma missão para descobrir mais informações sobre este herói ;)'
                        : props.heroInfo?.description}
                    </p>
                </div>
            </div>
        </>
    )
}

export default HeroImage;