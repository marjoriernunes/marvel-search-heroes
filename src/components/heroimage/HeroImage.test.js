import {render} from '@testing-library/react'
import HeroImage from './HeroImage'

test('renders correctly', () =>{
    const utils = render(<HeroImage/>)
    expect(utils.container).toMatchSnapshot()
});
