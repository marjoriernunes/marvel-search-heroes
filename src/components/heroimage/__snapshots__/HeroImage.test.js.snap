// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="header-hero-info"
  >
    <div
      class="gradient-overlay"
    >
      <img
        src="undefined/detail.undefined"
      />
      <span />
    </div>
    <div
      class="top-left"
    >
      <p
        class="hero-name"
      />
      <p
        class="hero-description"
      />
    </div>
  </div>
</div>
`;
